ttnn/cpp/ttnn/operations/creation.hpp

```diff
@@ -71,32 +71,41 @@ inline ttnn::Tensor full(
         memory_config.value_or(ttnn::DRAM_MEMORY_CONFIG));
 }
 
-inline ttnn::Tensor zeros(
-    const ttnn::Shape& shape,
-    const std::optional<DataType>& dtype = std::nullopt,
-    const std::optional<Layout>& layout = std::nullopt,
-    const std::optional<std::reference_wrapper<Device>>& device = std::nullopt,
-    const std::optional<MemoryConfig>& memory_config = std::nullopt) {
-    return full(shape, 0.0f, dtype, layout, device, memory_config);
-}
+namespace detail {
 
-inline ttnn::Tensor ones(
-    const ttnn::Shape& shape,
-    const std::optional<DataType>& dtype = std::nullopt,
-    const std::optional<Layout>& layout = std::nullopt,
-    const std::optional<std::reference_wrapper<Device>>& device = std::nullopt,
-    const std::optional<MemoryConfig>& memory_config = std::nullopt) {
-    return full(shape, 1.0f, dtype, layout, device, memory_config);
-}
+// Non-type template parameters (NTTPs) disallow floating point values
```

Comment:
nice it would be cool to move somewhere outside so we can use it in other places.

Reply:
@patrickroberts neat! 

https://en.cppreference.com/w/cpp/language/template_parameters#Non-type_template_parameter says floats are supposed to be allowed now in C++20 as NTTP but I get this error compiling with clang17:
```
<source>:6:17: error: sorry, non-type template argument of type 'double' is not yet supported
    6 | using bar = foo<2.0>;
```

Buuut, it compiles on clang18

Reply:
@cfjchu I agree this could be used more generally until we're on clang 18, but personally I'm an advocate of keeping code close to where it's used until it needs to be used elsewhere. Makes it a little easier to determine how much depends on it.